### ISSUE TITLE
[roslaunch] roslaunch-check doesn't respect if/unless

### ISSUE THREAD

**Issue by mikepurvis** [ros/ros_comm#993]
We're having a case where we have some launch files which optionally include other launch files based on an environment variable being set. These are failing the test created by `roslaunch_add_file_check`, since it doesn't respect the conditional and attempts to resolve all include elements without regard for context.

This can be reproduced trivially by a launch file such as this, which launches just fine, but will (wrongly, in my opinion) fail the roslaunch check:

```
<launch>
  <include if="0" file="does/not/exist.launch" />
</launch>
```

----

Interestingly, `roslaunch_add_file_check` totally does resolve `$(arg)`, `$(env)`, `$(optenv)`, etc, so this _really_ does seem to be an issue with just the conditionals.

**Comment by mikepurvis** [comment]
I believe the issue is here:

https://github.com/ros/ros_comm/blob/674263142a0be1124b273bdb1e0a79c3c956e259/tools/roslaunch/src/roslaunch/depends.py#L144-L148

It needs logic to explicitly check for if/unless, similar to how the args work:

https://github.com/ros/ros_comm/blob/674263142a0be1124b273bdb1e0a79c3c956e259/tools/roslaunch/src/roslaunch/depends.py#L99-L106

What's mysterious to me is that this works correctly for a normal launch but not for `roslaunch-check`. Is there some preprocessing going on that strips out tags based on if/unless attributes?

----

Looks like for a normal launch, this is handled by `xmlloader.py`, and the `@ifunless` decorator:

https://github.com/ros/ros_comm/blob/caab9fddc66dfa076dae6ac204c013fc49e2e67d/tools/roslaunch/src/roslaunch/xmlloader.py#L69-L96

@dirk-thomas It seems like there are some parallel implementation issues going on here, but I don't have the desire or energy to do a wholesale fixup of it. I'd prefer to fix the immediate bug as best I can and at least add a unit test which would help support a future refactoring. Unless you have some particular insight into what the relationship is between xmlloader and these other pieces?

**Comment by dirk-thomas** [comment]
Addressed by #998. Thank you!
